api/api/src/review_handler.rs

```diff
@@ -104,8 +152,18 @@ pub fn delete_review_handler(
     token: Result<JWT, (Status, Json<ResponseMessage>)>,
 ) -> Result<Json<ResponseMessage>, (Status, Json<ResponseMessage>)> {
     let token = token?;
+    let state = state.inner();
 
-    delete::delete_review(review_id, state, token)
+    match delete::delete_review(review_id, state, token) {
```

Comment:
This logic is repeated quite often; could this be made into a function/macro/`From` impl?